docs/publish.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# Publishes docs changes to the gh-pages branch.
+
+# abort script at first error:
+set -e
```

Comment:
Whoops, actually `publish.sh` was something I was still working on locally for auto-publishing doc artifacts to `gh-pages`. I'll take it out of this PR (but will make the change you suggested)